src/components/app-icon-button/AppIconButton.tsx

```diff
@@ -0,0 +1,21 @@
+import { IconButton, IconButtonProps } from '@mui/material'
+import { FC } from 'react'
+import { Link, To } from 'react-router-dom'
+
+interface AppIconButtonProps extends IconButtonProps {
+  to?: To
```

Comment:
How do you think, do we need to predict passing other link props here also because we are using only `to` prop here?
`react-router-dom` has following own props in `Link` (from docs):

```jsx
export interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "href"> {
    reloadDocument?: boolean;
    replace?: boolean;
    state?: any;
    preventScrollReset?: boolean;
    relative?: RelativeRoutingType;
    to: To;
    unstable_viewTransition?: boolean;
}
```

And do we need to take into account `NavLink` possible usage here or it is redundant?

Reply:
I added `LinkProps` to props type, so we can pass any `react-router-link` `Link` props. As for `NavLink`, it is not used anywhere in the project now, I don't think there would be many places where it can be used with Icon.